ddp-workspace/projects/ddp-dsm-ui/src/app/file-download/file-download.component.ts

```diff
@@ -21,47 +22,79 @@ export class FileDownloadComponent implements OnDestroy {
   downloading = false;
   CLEAN = 'CLEAN';
   subscription: Subscription;
+  subscription2: Subscription;
 
-  constructor(private dsmService: DSMService ) {
+  constructor( private dsmService: DSMService ) {
 
   }
 
   ngOnDestroy(): void {
     if (this.subscription != null) {
       this.subscription.unsubscribe();
     }
+    if (this.subscription2 != null) {
+      this.subscription2.unsubscribe();
+    }
   }
 
   downloadParticipantFile( file: File ): void {
     this.downloading = true;
-    if (file.scanResult !== this.CLEAN) {
+    if (!this.isFileClean( file )) {
       this.downloadMessage = 'Error - file has not passed scanning';
       this.downloading = false;
       return;
     }
     const realm = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
-    this.subscription = this.dsmService.downloadParticipantFile( file.fileName, file.bucket, file.blobName, realm, file.mimeType )
-      .subscribe( {
+    this.subscription = this.dsmService.getSignedUrl( this.participant.data.profile[ 'guid' ], file.fileName, file.bucket,
```

Comment:
it would be nice to limit number of arguments (for instance by using [TS object destructuring](https://www.typescriptlang.org/docs/handbook/variable-declarations.html#object-destructuring))

e.g. 
`this.dsmService.getSignedUrl( this.participant.data.profile[ 'guid' ], file.fileName, file.bucket, file.blobName, file.guid, realm)`

=> `this.dsmService.getSignedUrl( this.participant.data.profile[ 'guid' ], file, realm)`

Then in the service method:
 - either 
 ```
  public getSignedUrl( ddpParticipantId: string, file: File, realm: string) {
     const {fileName, bucket, blobName, guid} = file;
     ...
  }
```
- or short version  
`public getSignedUrl( ddpParticipantId: string, {fileName, bucket, blobName, guid}, realm: string) {...}`

- or short version with property renaming
`public getSignedUrl( ddpParticipantId: string, {fileName: newRenamedVariable, bucket, blobName, guid}, realm: string) {...}`

Reply:
For the header part, as you said, we can also just pass the file values, however we have to call this.buildQueryHeader( map ) to be able to get required HTTP headers in there as well.